supervisor/hyperpod.go

```diff
@@ -408,6 +409,20 @@ func (hp *HyperPod) createContainer(container, bundlePath, stdin, stdout, stderr
 
 	glog.Infof("createContainer() calls c.run(p)")
 	c.run(p)
+	alarm := time.After(5 * time.Second)
+	for {
+		if pid := c.ownerPod.getNsPid(); pid > 0 {
+			p.ProcId = pid
+			break
+		}
+
+		select {
+		case <-alarm:
+			glog.V(3).Infof("Timeout while waiting for container's pid")
+		default:
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
```

Comment:
The line "c.run(p)" above this piece of code is put into a goroutine, the creating process is also put into the goroutine, we have to wait because nothing is done when this "waiting" is executed.